artifactory.py

```diff
@@ -997,11 +997,30 @@ def deploy(
         parameters=None,
         explode_archive=None,
         explode_archive_atomic=None,
+        checksum=None,
+        by_checksum=None,
```

Comment:
```suggestion
        by_checksum=False,
```

Reply:
Hi @beliaev-maksim 
Thanks for your time and effort.

I am not trying to go against the code of conduct in existing code, like what we do about `md5`, `sha1`, `sha256`, `explode_archive` and `explode_archive_atomic`.

Besides boolean parameter in `deploy` in `_ArtifactoryAccessor`, this issue exists in other function as well. If we change `by_checksum=None` to `by_checksum=False`, do we have a plan to change other parameters either?

If the answer is 'yes', it's fine for me to make this change.

Reply:
you can change `explode_archive` and `explode_archive_atomic` to False as well. This will keep backwards compatibility.

If None is used, then by PEP guides is recommended to use a direct comparison "if val is None", and we do not do this in the code, right?
then bool values are more logical

Reply:
> you can change `explode_archive` and `explode_archive_atomic` to False as well. This will keep backwards compatibility.

As `md5`, `sha1`, `sha256`, `explode_archive` and `explode_archive_atomic` are all in same type, `bool`, do you mean only change `explode_archive` and `explode_archive_atomic` to `False`? Is there some special reasons changing only two of them, and keeping the rest as before? Do we have a plan to change bool parameter default value to False/True instead of None in other functions?


> If None is used, then by PEP guides is recommended to use a direct comparison "if val is None", and we do not do this in the code, right?
> then bool values are more logical

It does make sense. Thanks for you clarification

Reply:
@zhan9san 
potentially we can consider changing other functions. However, it is too dangerous to change blindly without code review, thus, I cannot reply now about other methods

Reply:
Fixed